esx_service/vmdk_ops_test.py

```diff
@@ -252,7 +255,156 @@ def test_failure(self):
+    volNamePre = "vol_UnitTest_Attach"
+    vm_name = 'test-vm'
+    max_vol_count = 60
+    
+    def setUp(self):
+        """ Setup run before each test """
+        logging.debug("VMDKAttachDetachTest setUp path =%s", path)
+        self.cleanup()
+        
+        writeable_ds_path = 'datastore1'
+        if writeable_ds_path:
+            # get service_instance, and create a VM
+            if not vmdk_ops.si:
+                vmdk_ops.connectLocal()
+
+            self.create_vm(vmdk_ops.si, self.vm_name, writeable_ds_path)
+
+            # create max_vol_count+1 VMDK files
+            for id in range(1, self.max_vol_count+2):
+                volName = 'VmdkAttachDetachTestVol' + str(id)
+                fullpath = os.path.join(path, volName + '.vmdk')
+                self.assertEqual(None,
+                                 vmdk_ops.createVMDK(vm_name='test-vm',
+                                                     vmdk_path=fullpath,
+                                                     vol_name=volName))
+                
+    def tearDown(self):
+        """ Cleanup after each test """
+        logging.debug("VMDKAttachDetachTest tearDown path")
+        self.cleanup()
+    
+    def create_vm(self, si, vm_name, datastore):
+        content = si.RetrieveContent()
+        datacenter = content.rootFolder.childEntity[0]
+        vmfolder = datacenter.vmFolder
+        hosts = datacenter.hostFolder.childEntity
+        resource_pool = hosts[0].resourcePool
+
+        datastore_path = '[' + datastore + '] ' + vm_name
+
+        # bare minimum VM shell, no disks. Feel free to edit
+        vmx_file = vim.vm.FileInfo(logDirectory=None,
+                                   snapshotDirectory=None,
+                                   suspendDirectory=None,
+                                   vmPathName=datastore_path)
+
+
+        config = vim.vm.ConfigSpec( 
+                                name=vm_name, 
+                                memoryMB=128, 
+                                numCPUs=1,
+                                files=vmx_file, 
+                                guestId='rhel5_64Guest', 
+                                version='vmx-11'
+                              )
+
+        #logging.debug("Createing VM %s", vm_name)
+        task = vmfolder.CreateVM_Task(config=config, pool=resource_pool)
+        vmdk_ops.wait_for_tasks(si, [task])
+
+        vm = [d for d in si.content.rootFolder.childEntity[0].vmFolder.childEntity if d.config.name == vm_name]
+        if (vm):
+            logging.debug("Found: VM %s", vm_name)
+            if vm[0].runtime.powerState == vim.VirtualMachinePowerState.poweredOff:
+                logging.debug("Attempting to power on %s", vm_name)
+                task = vm[0].PowerOnVM_Task()
+                vmdk_ops.wait_for_tasks(si, [task])
+    
+    def remove_vm(self, si, vm_name):
+
+        vm = [d for d in si.content.rootFolder.childEntity[0].vmFolder.childEntity if d.config.name == vm_name]
```

Comment:
If the VM is running then vmdk_ops:findVmByUuid() will get the vm object, this loop isn't needed here.